src/csharp/Microsoft.Spark/ML/Feature/FeatureHasher.cs

```diff
@@ -0,0 +1,145 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Spark.Interop;
+using Microsoft.Spark.Interop.Ipc;
+using Microsoft.Spark.Sql;
+using Microsoft.Spark.Sql.Types;
+
+namespace Microsoft.Spark.ML.Feature
+{
+    public class FeatureHasher: FeatureBase<FeatureHasher>, IJvmObjectReferenceProvider
+    {
+        private static readonly string s_featureHasherClassName = 
+            "org.apache.spark.ml.feature.FeatureHasher";
+        
+        internal FeatureHasher() : base(s_featureHasherClassName)
+        {
+        }
+
+        internal FeatureHasher(string uid) : base(s_featureHasherClassName, uid)
+        {
+        }
+
+        internal FeatureHasher(JvmObjectReference jvmObject) : base(jvmObject)
+        {
+        }
+
+        JvmObjectReference IJvmObjectReferenceProvider.Reference => _jvmObject;
+        
+        /// <summary>
+        /// Loads the <see cref="FeatureHasher"/> that was previously saved using Save.
+        /// </summary>
+        /// <param name="path">
+        /// The path the previous <see cref="FeatureHasher"/> was saved to.
+        /// </param>
+        /// <returns>New <see cref="FeatureHasher"/> object</returns>
+        public static FeatureHasher Load(string path) =>
+            WrapAsFeatureHasher(
+                SparkEnvironment.JvmBridge.CallStaticJavaMethod(
+                    s_featureHasherClassName,"load", path));
+        
+        /// <summary>
+        /// Gets a list of the columns which have been specified as categorical columns.
+        /// </summary>
+        /// <returns>List of categorical columns, set by SetCategoricalCols</returns>
+        public IEnumerable<string> GetCategoricalCols() => 
+            ((string[])_jvmObject.Invoke("getCategoricalCols")).ToList();
+        
+        /// <summary>
+        /// Marks columns as categorical columns.
+        /// </summary>
+        /// <param name="value">List of column names to mark as categorical columns</param>
+        /// <returns>New <see cref="FeatureHasher"/> object</returns>
+        public FeatureHasher SetCategoricalCols(IEnumerable<string> value) => 
+            WrapAsFeatureHasher(_jvmObject.Invoke("setCategoricalCols", value));
+        
+        /// <summary>
+        /// Gets the columns that the <see cref="FeatureHasher"/> should read from and convert into
+        /// hashes. This would have been set by SetInputCol.
+        /// </summary>
+        /// <returns>IEnumerable&lt;string&gt;, the input columns</returns>
```

Comment:
I am guessing `&lt` and `&gt` have been added for the benefit of proper rendering on the API docs. However, if that is the case, I wonder if we should be consistent about this, for instance, in line 48.